core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -1203,4 +1204,47 @@ default StreamMessage<T> subscribeOn(EventExecutor eventExecutor) {
         requireNonNull(eventExecutor, "eventExecutor");
         return new SubscribeOnStreamMessage<>(this, eventExecutor);
     }
+
+    /**
+     * Configures a timeout for the stream based on the specified duration.
+     * The default timeout mode is {@link StreamTimeoutMode#UNTIL_NEXT}.
+     *
+     * <p>Example usage:
+     * <pre>{@code
+     * StreamMessage<String> stream = ...;
+     * StreamMessage<String> timeoutStream = stream.timeout(Duration.ofSeconds(10));
+     * }</pre>
+     *
+     * @param timeoutDuration the duration before a timeout occurs
+     * @return a new {@link TimeoutStreamMessage} with the specified timeout duration and default mode
+     */
+    @UnstableApi
+    default StreamMessage<T> timeout(Duration timeoutDuration) {
+        requireNonNull(timeoutDuration, "timeoutDuration");
+        return timeout(timeoutDuration, StreamTimeoutMode.UNTIL_NEXT);
+    }
+
+    /**
+     * Configures a timeout for the stream based on the specified duration and mode.
+     * Internally, it creates and returns a {@link TimeoutStreamMessage} with the specified parameters.
```

Comment:
It would be better to write the behavior and contract instead of internal details.
```suggestion
```